roles/satellite-clone/files/disassociate_capsules.rb

```diff
@@ -24,30 +24,21 @@ def capsule_lce_args(action, capsule_id, env)
   "--csv capsule content #{action}-lifecycle-environment --id #{capsule_id} --lifecycle-environment-id #{env}"
 end
 
-external_capsules = []
 external_capsule_ids = get_info_from_hammer("--csv capsule list --search 'feature = \"Pulp Node\"'")
-if external_capsule_ids.empty?
-  STDOUT.puts "There are no external capsules to disassociate."
-else
-  external_capsule_ids.split("\n").each do |id|
-    external_capsules << {
-      id: id,
-      lifecycle_environments: get_info_from_hammer("--csv capsule content lifecycle-environments --id #{id}").split("\n")
-    }
-  end
+STDOUT.puts "There are no external capsules to disassociate." if external_capsule_ids.empty?
```

Comment:
Removing the if/else this way does change the behavior. Now it will also print the long message at the end of the script if it's empty.

Reply:
Note the inline conditional which I added here:

```ruby
STDOUT.puts "All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite " \
            "and any interference with existing infrastructure. To reverse these changes, run the following commands," \
            " making sure to replace the credentials with your own." unless reverse_commands.empty?
```

But I had some concern about readability when I wrote it this way. Would it be better style to wrap that in an if, since the string is split across several lines?

Reply:
I think that the `if/else` nesting that was there was easier to follow.

Reply:
I updated it with a block `unless reverse_commands.empty?` conditional for the final output.

Technically this does fix an issue that existed previously where the long final output message would be displayed, with no following reverse_commands, if there were external capsules but the LCEs were already disassociated (hence `external_capsule_ids` would be populated while `reverse_commands` would be empty). That is kind of an edge case but the logic didn't quite seem right and `unless reverse_commands.empty?` more accurately captures what it should really care about.

A side effect of that fix is that I opted to leave the `external_capsules = ...` and `reverse_commands = ...` definition blocks at the top level rather than under an `if/else`, since they will wind up empty anyway if the arrays they are built from are also empty and `if external_capsule_ids.empty?` was never really the correct conditional for every bit of code underneath it.

Let me know what you think, thanks!

Reply:
Looking at it some more, I think ` STDOUT.puts "There are no external capsules to disassociate." if external_capsule_ids.empty?` also belongs in the same block down below for similar reasons. I think the correct form should be:

```ruby
if reverse_commands.empty?
  STDOUT.puts "There were no Lifecycle Environments to disassociate from external capsules."
else
  STDOUT.puts "All Capsules are unassociated with any lifecycle environments. This is to avoid any syncing errors with your original Satellite " \
              "and any interference with existing infrastructure. To reverse these changes, run the following commands," \
              " making sure to replace the credentials with your own."
   reverse_commands.each { |command| STDOUT.puts command }
 end
 ```